backend/src/main/java/de/bund/digitalservice/ris/caselaw/adapter/DatabaseDocumentationOfficeUserGroupService.java

```diff
@@ -0,0 +1,155 @@
+package de.bund.digitalservice.ris.caselaw.adapter;
+
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DatabaseDocumentationOfficeRepository;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DatabaseDocumentationOfficeUserGroupRepository;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DocumentationOfficeDTO;
+import de.bund.digitalservice.ris.caselaw.adapter.database.jpa.DocumentationOfficeUserGroupDTO;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOffice;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOfficeUserGroup;
+import de.bund.digitalservice.ris.caselaw.domain.DocumentationOfficeUserGroupService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DatabaseDocumentationOfficeUserGroupService
+    implements DocumentationOfficeUserGroupService {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(DatabaseDocumentationOfficeUserGroupService.class);
+  private final DatabaseDocumentationOfficeUserGroupRepository repository;
+  private final DatabaseDocumentationOfficeRepository documentationOfficeRepository;
+  private List<DocumentationOfficeUserGroupDTO> documentationOfficeUserGroups;
+
+  private final List<DocumentationOfficeConfigUserGroup> userGroupsFromConfig;
+
+  public DatabaseDocumentationOfficeUserGroupService(
+      DatabaseDocumentationOfficeUserGroupRepository repository,
+      DatabaseDocumentationOfficeRepository documentationOfficeRepository,
+      List<DocumentationOfficeConfigUserGroup> documentationOfficeConfigUserGroups) {
+    this.repository = repository;
+    this.documentationOfficeRepository = documentationOfficeRepository;
+    this.documentationOfficeUserGroups = new ArrayList<>();
+    this.userGroupsFromConfig = documentationOfficeConfigUserGroups;
+  }
+
+  /**
+   * On application start, we want to
+   *
+   * <ol>
+   *   <li>sync the user group database with the statically configured list of user groups
+   *   <li>load the user groups into memory for performance reasons (-> right checks)
+   * </ol>
+   */
+  @EventListener
+  public void onApplicationEvent(ContextRefreshedEvent event) {
+    this.documentationOfficeUserGroups = this.repository.findAll();
+    var docOffices = this.documentationOfficeRepository.findAll();
+
+    var userGroupsToBeDeleted =
+        documentationOfficeUserGroups.stream().filter(this::groupIsNotInConfig).toList();
+    if (!userGroupsToBeDeleted.isEmpty()) {
+      LOGGER.info("Deleting {} doc office user groups", userGroupsToBeDeleted.size());
```

Comment:
I would argue that without specifying the deleted user groups this log might not be very useful, is it? Does anything speak against logging the groups?